src/graph_notebook/magics/streams.py

```diff
@@ -0,0 +1,225 @@
+import re
+import json
+import urllib.request
+import urllib.error
+import ipywidgets as widgets
+import queue
+from IPython.display import display, HTML, clear_output
+from graph_notebook.neptune.client import STREAM_AT, STREAM_AFTER, STREAM_TRIM, STREAM_EXCEPTION_NOT_FOUND
+class EventId:
+    def __init__(self, commit_num=1, op_num=1):
+        self.commit_num = int(commit_num)
+        self.op_num = int(op_num)
+        
+    
+    def update(self, event_id):
+        if event_id is not None:
+            self.commit_num = event_id.commit_num
+            self.op_num = event_id.op_num
+            
+    def duplicate(self):
+        return EventId(self.commit_num, self.op_num)
+    
+    def value(self):
+        return '{}/{}'.format(self.commit_num, self.op_num)
+
+class StreamClient:
+    
+    def __init__(self, wb_client, uri_with_port, limit=10):
+        self.wb_client = wb_client
+        self.uri_with_port = uri_with_port
+        self.limit=limit
+    
+    
+    def get_events(self, language, event_id, iterator):
+        try:
+            jsonresponse = self.wb_client.stream(self.__stream_uri(language),
+                                                 iteratorType = iterator,
+                                                 commitNum = event_id.commit_num,
+                                                 opNum = event_id.op_num,
+                                                 limit=self.limit)
+                                                 
+            if 'records' in jsonresponse:
+                records = jsonresponse['records']
+                first_event = EventId(records[0]['eventId']['commitNum'], records[0]['eventId']['opNum'])
+                last_event = EventId(jsonresponse['lastEventId']['commitNum'], jsonresponse['lastEventId']['opNum'])
+
+                return (records, first_event, last_event)
+            else:
+                return ([], None, None)
+                
+        except:
+            return ([], None, None)
+        
+    def __parse_last_commit_num(self, msg):
+        results = re.findall("\d+", msg)      
+        return None if not results else results[0]
+    
+    def __stream_uri(self, language):
+        uri = f'{self.uri_with_port}/{language.lower()}/stream/'
+        return uri
+    
+    # TODO: Revisit this logic if the Neptune Stream API adds support for querying this
+    # directly.
+    def get_last_commit_num(self, language):
+           
+        commit_num = 1000000000
```

Comment:
Is this number going to be large enough, maybe use sys.maxsize()?

Reply:
That did occur to me too the other day although I was thinking about using (2^31)-1 rather than `sys.maxint` which is a huge number I need to check that the REST API can accept (9223372036854775807) - a.k.a. (2^63)-1. @iansrobinson any thoughts?

Reply:
This is rewritten now to use 2^^63-1